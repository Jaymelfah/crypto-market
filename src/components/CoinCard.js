import React from 'react';
import PropTypes from 'prop-types';

const CoinCard = (props) => {
  const { coins } = props;

  return (

    <div className="coin-card flex">
      <h4>{coins.name}</h4>
      <img src={coins.image} alt={coins.name} />
      <p>
        Market Ranking:
        {' '}
        {coins.rank}
      </p>
      <p>
        Price:
        {' '}
        $
        {' '}
        {coins.currentPrice}
      </p>
    </div>

  );
};

CoinCard.propTypes = {
  coins: PropTypes.array,
}.isRequired;

export default CoinCard;
