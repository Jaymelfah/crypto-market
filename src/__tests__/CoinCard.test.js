import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CoinCard from '../components/CoinCard';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';

describe('Coin Card Rendering', () => {
  const coins = {
    name: 'Ethereum',
    image: '',
    rank: 2,
    currentPrice: '234',
  };
  test('Check if an image is rendered', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <CoinCard coins={coins} />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    const image = screen.getByRole('img');
    expect(image).toBeTruthy();
  });

  test('Render coin cards on the page', () => {
    const coinCard = render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <CoinCard coins={coins} />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    );
    expect(coinCard).toMatchSnapshot();
  });
});
