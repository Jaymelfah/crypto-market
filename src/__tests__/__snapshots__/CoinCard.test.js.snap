// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Card Rendering Render coin cards on the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coin-card flex"
      >
        <h4>
          Ethereum
        </h4>
        <img
          alt="Ethereum"
          src=""
        />
        <p>
          Market Ranking:
           
          2
        </p>
        <p>
          Price:
           
          $
           
          234
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coin-card flex"
    >
      <h4>
        Ethereum
      </h4>
      <img
        alt="Ethereum"
        src=""
      />
      <p>
        Market Ranking:
         
        2
      </p>
      <p>
        Price:
         
        $
         
        234
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
